docs/proposals/20190708-MachineConfig.md

```diff
@@ -0,0 +1,293 @@
+# Proposal: Machine Configuration in Talos v0.2.0 and Beyond
+
+Author(s): Spencer Smith (@rsmitty)
+
+## Abstract
+
+This proposal will outline how we'll handle the passing of machine configurations in Talos, as we're aiming to simplify the options and keep the user from having to worry about the minutiae of kubeadm and all of the options.
+
+## Background
+
+I think the easiest way to background this is to take a look at each of the machine configs that we currently have templates for. When looking at these files, it's somewhat self-explanatory on what is available to tweak, but it also gives a good idea of what is similar between the three types of Talos nodes: init (the first master), control plane (any other masters), and workers. I've also appended some additional fields that we use for certain platforms like Packet.
```

Comment:
I think it is worth pointing out that we are going explicitly away from the term "user data" in order to distinguish ourselves from `cloud-init`.